File: Ruby/iterate-array-hash.rb
@country_data =   [
      {capital: 'London', country: 'UK', country_code: 'GB'},
      {capital: 'Paris', country: 'France', country_code: 'FR'},
      {capital: 'Madrid', country: 'Spain', country_code: 'ES'},
      {capital: 'Rome', country: 'Italy', country_code: 'IT'}
    ]

# Print all in the format: Paris, France [FR]
# Iterate with each and interpolate variables
@country_data.each do |country|
  puts "#{country.values[0]}, #{country.values[1]} [#{country.values[2]}]"
end

# Print the former last country of the array
puts "Former last country is #{@country_data[-2][:country]}."

def get_country_code(city) # Paris
  @country_data.each do |hash|
    return hash[:country_code] if hash[:capital] == city
  end
  "?"
end

puts get_country_code('Paris') # Should return 'FR'
puts get_country_code('New York') # Should return '?'

File: JS/iterate-array-hash.js
const countryData = [
      {capital: 'London', country: 'UK', country_code: 'GB'},
      {capital: 'Paris', country: 'France', country_code: 'FR'},
      {capital: 'Madrid', country: 'Spain', country_code: 'ES'},
      {capital: 'Rome', country: 'Italy', country_code: 'IT'}
    ];

// Print all in the format: Paris, France [FR]
countryData.forEach((country) => { console.log(`${country.capital}, ${country.country} [${country.country_code}]`) })

// Print the former last country of the array
let formerLast = countryData[countryData.length - 2].country;
console.log(`Former last country of the array is ${formerLast}.`);

// getCountryCode('Paris') # Should return 'FR'
// getCountryCode('New York') # Should return '?'

const getCountryCode = (city) => {
  let code = countryData.forEach((country) => {
    if ( city === country.capital) { console.log(country.country_code) };
  });
  return code || "?";
};

console.log(getCountryCode('Paris'));// Should return 'FR'
// console.log(getCountryCode('New York'));// Should return '?'
